homeassistant/components/hardware/websocket_api.py

```diff
@@ -45,3 +72,48 @@ async def ws_info(
                 hardware_info.append(asdict(platform.async_info(hass)))
 
     connection.send_result(msg["id"], {"hardware": hardware_info})
+
+
+@websocket_api.websocket_command(
+    {
+        vol.Required("type"): "hardware/subscribe_system_status",
+    }
+)
+@websocket_api.async_response
+async def ws_subscribe_system_status(hass, connection, msg):
+    """Subscribe to system status updates."""
+
+    system_status: SystemStatus = hass.data[DOMAIN]["system_status"]
+
+    @callback
+    def async_update_status(now: datetime) -> None:
+        cpu_percentage = round(system_status.psutil.cpu_percent(interval=None))
+        virtual_memory = system_status.psutil.virtual_memory()
+        json_msg = {
```

Comment:
Can we add the date time this is sent? I can add it in the frontend but may be best to get the timing from when its polled in the backend